src/modules/cpu/kernel/spectrogram.hpp

```diff
@@ -0,0 +1,234 @@
+    Rpp32u numThreads = handle.GetNumThreads();
+
+    // Get windows output
+    omp_set_dynamic(0);
+#pragma omp parallel for num_threads(numThreads)
+    for (Rpp32s batchCount = 0; batchCount < srcDescPtr->n; batchCount++)
+    {
+        Rpp32f *srcPtrTemp = srcPtr + batchCount * srcDescPtr->strides.nStride;
+        Rpp32f *dstPtrTemp = dstPtr + batchCount * dstDescPtr->strides.nStride;
+        Rpp32s bufferLength = srcLengthTensor[batchCount];
+        Rpp32s numWindows = get_num_windows(bufferLength, windowLength, windowStep, centerWindows);
+        Rpp32f windowOutput[numWindows * nfft];
+        std::fill_n(windowOutput, numWindows * nfft, 0);
+        for (Rpp32s w = 0; w < numWindows; w++)
+        {
+            Rpp32s windowStart = w * windowStep - windowCenterOffset;
+            Rpp32f *windowOutputTemp = windowOutput + (w * nfft);
+            // Pad when either windowStart less than zero or length greater than input srclength
+            if (windowStart < 0 || (windowStart + windowLength) > bufferLength)
+            {
+                for (Rpp32s t = 0; t < windowLength; t++)
+                {
+                    Rpp32s inIdx = windowStart + t;
+                    if (reflectPadding)
+                    {
+                        inIdx = get_idx_reflect(inIdx, 0, bufferLength);
+                        *windowOutputTemp++ = windowFn[t] * srcPtrTemp[inIdx];
+                    }
+                    else
+                    {
+                        if (inIdx >= 0 && inIdx < bufferLength)
+                            *windowOutputTemp++ = windowFn[t] * srcPtrTemp[inIdx];
+                        else
+                            *windowOutputTemp++ = 0;
+                    }
+                }
+            }
+            else
+            {
+                Rpp32f *srcPtrWindowTemp = srcPtrTemp + windowStart;
+                Rpp32f *windowFnTemp = windowFn;
+                Rpp32s t = 0;
+                for (; t < alignedWindowLength; t += 8)
+                {
+                    __m256 pSrc, pWindowFn;
+                    pSrc = _mm256_loadu_ps(srcPtrWindowTemp);
+                    pWindowFn = _mm256_loadu_ps(windowFnTemp);
+                    pSrc = _mm256_mul_ps(pSrc, pWindowFn);
+                    _mm256_storeu_ps(windowOutputTemp, pSrc);
+                    srcPtrWindowTemp += 8;
+                    windowFnTemp += 8;
+                    windowOutputTemp += 8;
+                }
+                for (; t < windowLength; t++)
+                    *windowOutputTemp++ = (*windowFnTemp++) * (*srcPtrWindowTemp++);
+            }
+        }
+
+        // Generate FFT output
+        ffts_plan_t *p;
+        if(useRealImpl)
+            p = ffts_init_1d_real(nfft, FFTS_FORWARD);
+        else
+            p = ffts_init_1d(nfft, FFTS_FORWARD);
+
+        if (!p)
+        {
+            printf("FFT Plan is unsupported. Exiting the code\n");
```

Comment:
Remove print statement. Create an error code RPP_ERROR_FFT_PLAN_UNSUPPORTED and return that. In the cpp API file, return whats returned here.

Reply:
We won't be able to judge anything on ffts_plan_t in cpp API file, please correct me if am wrong.
p is based on 
```
if(useRealImpl)
            p = ffts_init_1d_real(nfft, FFTS_FORWARD);
        else
            p = ffts_init_1d(nfft, FFTS_FORWARD);
```

Reply:
Removed print statement and returned error code RPP_ERROR_FFT_PLAN_UNSUPPORTED

Reply:
Adding return inside openMP segment leads to error.

Reply:
@snehaa8 We ideally don't want print statements inside lib code. Lets find a work around, and you can issue a separate PR.